core/conn/jdbc_type2/native/SqlInterface.cpp

```diff
@@ -246,10 +246,14 @@ SQLRETURN GetJDBCValues( SQLItemDesc_def *SQLItemDesc,
 		SQLItemDesc->signType      = TRUE;
 		break;
 	case SQLTYPECODE_DATETIME:
+		// Calculate the length based on YYYY/MM/DD HH:MM:SS.ffffff
+        SQLItemDesc->maxLen = 20 + 6;
```

Comment:
JDBC T2 use VARCHAR to handle DATE type.So , we need set the maxLen in T2. 

Maybe a better fix is using the binary time format. But I need more time to investigate how to use binary date format from SQLCLI .  So I want to submit this little fix to make the MT-DCS can run with windows ODBC.